src/components/header/EmptyThreadIndicator.js

```diff
@@ -0,0 +1,158 @@
+import type { Milliseconds, StartEndRange } from '../../types/units';
+
+import './EmptyThreadIndicator.css';
+
+type Props = {
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +width: number,
+  +thread: Thread,
+  +unfilteredSamplesRange: StartEndRange | null,
+};
+
+type State = {
+  hoveredItem: *,
+  mouseX: number,
+  mouseY: number,
+};
+
+class EmptyThreadIndicator extends PureComponent<Props, State> {
+  _canvas: HTMLCanvasElement | null;
+  _requestedAnimationFrame: boolean | null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      hoveredItem: null,
+      mouseX: 0,
+      mouseY: 0,
+    };
+  }
+
+  _getIndicatorPositions(): {
+    startup?: Object,
+    shutdown?: Object,
+    emptyBufferStart?: Object,
+    emptyBufferEnd?: Object,
+  } {
+    const {
+      rangeStart,
+      rangeEnd,
+      width,
+      unfilteredSamplesRange,
+      thread: {
+        processShutdownTime,
+        processStartupTime,
+        registerTime,
+        unregisterTime,
+      },
+    } = this.props;
+    const style = {};
+    const rangeLength = rangeEnd - rangeStart;
+    const xPixelsPerMs = width / rangeLength;
+
+    // Compute the thread start and end absolute times.
+    const threadStartTime = processStartupTime + registerTime;
+    let threadShutdownTime = null;
+    if (unregisterTime !== null) {
+      threadShutdownTime = processStartupTime + unregisterTime;
+    } else if (processShutdownTime !== null) {
+      threadShutdownTime = processShutdownTime;
+    }
+
+    // Did this thread startup in this time range?
+    if (threadStartTime > rangeStart) {
+      style.startup = {
+        left: 0,
+        width: (processStartupTime - rangeStart) * xPixelsPerMs,
+      };
+    }
+
+    // Did this thread shutdown in this time range?
+    if (threadShutdownTime !== null && threadShutdownTime < rangeEnd) {
+      style.shutdown = {
+        right: 0,
+        width: (rangeEnd - threadShutdownTime) * xPixelsPerMs,
+      };
+    }
+
+    if (unfilteredSamplesRange !== null) {
+      if (unfilteredSamplesRange.start > rangeStart) {
+        const startMilliseconds = Math.max(0, processStartupTime - rangeStart);
+        style.emptyBufferStart = {
+          left: startMilliseconds * xPixelsPerMs,
+          right:
+            width - (unfilteredSamplesRange.start - rangeStart) * xPixelsPerMs,
+        };
+      }
+
+      if (unfilteredSamplesRange.end < rangeEnd) {
+        const endMilliseconds =
+          threadShutdownTime === null
+            ? rangeEnd
+            : Math.min(rangeEnd, threadShutdownTime);
+        style.emptyBufferEnd = {
+          left:
+            Math.max(0, unfilteredSamplesRange.end - rangeStart) * xPixelsPerMs,
+          width: (endMilliseconds - rangeStart) * xPixelsPerMs,
+        };
+      }
+    }
+    return style;
+  }
+
+  render() {
+    const { hoveredItem, mouseX, mouseY } = this.state;
+    const shouldShowTooltip = false;
```

Comment:
I believe this isn't finished ? :)

Reply:
Yeah, that's what I get for being in a hurry.